entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryConverter.java

```diff
@@ -337,10 +343,22 @@ private AttributeFilter convertToAttributeFilter(
                                         .build())))
                 .build());
         break;
+      case UNRECOGNIZED:
+      default:
+        throw new IllegalArgumentException(
+            String.format("Unrecognized value type %s for literal %s", valueType, literal));
     }
     return builder;
   }
 
+  private static boolean isValueTypeSet(org.hypertrace.entity.query.service.v1.Value value) {
+    return value.hasField(
```

Comment:
Are we trying to distinguish between unset and set to default? I'm fairly sure this won't work and almost positive we can't _guarantee_ it will work. Proto is free to represent a value in whatever way it can for transport, we can't assume it would represent set to default and unset any differently.